### DISCUSSION TITLE
How can i get 'Router' and 'window' in 'Constructor' of class component in nextjs ?

### DISCUSSION THREAD

**Discussion by bhaveshpatel200** [vercel/next.js#13492] [Help]
How can i get 'Router' and 'window' in 'Constructor' of class component in nextjs ?

**Comment by Timer** [comment]
Since Next.js prerenders to HTML to ensure your app is fast, you cannot always access `window`.

You can only use `window` in `useEffect` (hooks) or `componentDidMount` (class components).

**Comment by bhaveshpatel200** [comment]
If i will check user is authenticated or not in 'componentDidMount' , then if user is authenticated then page fluctuate while get auth user details in 'componentDidMount' using api.

ex. user is not authenticated, then user access any authorized page 'home' at url, then first render home page and then know user is not authenticated and we will redirect to login page.
here we need to prevent to render 'Home' page when user is not authenticated, so.

there is any way where we can check user authentication, if then can you please saher example.

this is we are doing in angular using 'Authgaurd'.
i want to same needed nextjs.

Thanks.
